compiler/src/template-api/Table.js

```diff
@@ -0,0 +1,318 @@
+            typeof table_params.width === "object"
+                ? table_params.width
+                : new Array(fields.length).fill(
+                      table_params.width / fields.length
+                  );
+        var cell_X = [x];
+        for (var i = 1; i < fields.length; i++) {
+            cell_X.push(cell_X[i - 1] + cell_W[i - 1]);
+        }
+        var cell_H = table_params.cell_h;
+
+        var th_params = table_params.heads;
+        var th_X = cell_X;
+        var th_Y = y;
+        var th_W = cell_W;
+        var th_H = th_params.th_h;
+        var th_Names = th_params.names || fields;
+
+        var table = g
+            .append("g")
+            .attr("width", table_params.width)
+            .attr("height", 500) //TODO: not hard code it
+            .attr("id", "kyrix_table");
+
+        var ths = table
+            .append("g")
+            .attr("class", "thead")
+            .attr("transform", function(d, i) {
+                return "translate(0," + th_Y + ")";
+            });
+
+        var trs = table
+            .selectAll(".tr")
+            .data(data)
+            .enter()
+            .append("g")
+            .attr("class", "tr")
+            .attr("transform", function(d) {
+                return "translate(0," + (d["y"] - cell_H / 2) + ")";
+            });
+
+        for (var index = 0; index < fields.length; index++) {
+            var th = ths
+                .append("g")
+                .attr("class", "th " + th_Names[index])
+                .attr("transform", "translate(" + th_X[index] + ",0)");
+
+            trs.each(function(d, i, nodes) {
+                var tr = d3.select(this);
+                var cell = tr
+                    .append("g")
+                    .attr("class", function(d, i) {
+                        return "td cell " + fields[index];
+                    })
+                    .attr("transform", function(d, i) {
+                        return "translate(" + cell_X[index] + ",0)";
+                    })
+                    .attr("width", cell_W[index])
+                    .attr("height", cell_H);
+
+                cell.append("rect")
+                    .attr("class", "rect bg")
+                    .attr("width", cell_W[index])
+                    .attr("height", cell_H);
+                cell.append("text")
+                    .text(function(d) {
+                        return d[fields[index]];
+                    })
+                    .attr("dx", cell_W[index] / 2)
+                    .attr("y", cell_H / 2)
+                    .attr("dy", "0.5em");
+            });
+
+            th.append("rect")
+                .attr("width", th_W[index])
+                .attr("height", th_H);
+
+            th.append("text")
+                .text(th_Names[index])
+                .attr("dx", th_W[index] / 2)
+                .attr("y", th_H / 2)
+                .attr("dy", "0.5em");
+        }
+    }
+}
+
+function getBodyStringOfFunction(func) {
```

Comment:
This function seems to be used in every template, so let's move it to `template-api/Renderers.js`